builder/base_image.go

```diff
@@ -25,53 +26,89 @@ func NewBaseImageBuilder(baseImage string) *BaseImageBuilder {
 	}
 }
 
-// CreateDockerfileDir generates a Dockerfile and assets in the targetDir
-func (b *BaseImageBuilder) CreateDockerfileDir(targetDir, configginTarballPath string) error {
-	if err := os.MkdirAll(targetDir, 0755); err != nil {
-		return err
-	}
-
-	dockerfilePath := filepath.Join(targetDir, "Dockerfile")
-	dockerfileContents, err := b.generateDockerfile()
-	if err != nil {
-		return err
-	}
-
-	if err := ioutil.WriteFile(dockerfilePath, dockerfileContents, 0644); err != nil {
-		return err
-	}
-
-	if err := b.unpackConfiggin(targetDir, configginTarballPath); err != nil {
-		return err
-	}
-
-	if err := dockerfiles.RestoreAsset(targetDir, "monitrc.erb"); err != nil {
-		return err
-	}
-	if err := os.Chmod(filepath.Join(targetDir, "monitrc.erb"), 0600); err != nil {
-		return err
-	}
-
-	if err := dockerfiles.RestoreAssets(targetDir, "rsyslog_conf"); err != nil {
-		return err
-	}
-
-	return nil
-}
-
-func (b *BaseImageBuilder) unpackConfiggin(targetDir, configginTarballPath string) error {
-
-	configginDir := filepath.Join(targetDir, "configgin")
-
-	if err := os.MkdirAll(configginDir, 0755); err != nil {
-		return err
-	}
-
-	if err := extractor.NewTgz().Extract(configginTarballPath, configginDir); err != nil {
-		return err
-	}
-
-	return nil
+// CreateDockerStream generates a Dockerfile and assets in the targetDir
+func (b *BaseImageBuilder) CreateDockerStream(configginTarballPath string) (io.ReadCloser, <-chan error) {
```

Comment:
From further discussion, we made it all the docker wrapper layer's job, and pass in a `func (*tar.Writer) error` instead.